src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2979,85 +2974,73 @@ class CgroupUtils:
-                    # look at the server jobs list instead of the moms
+                try:
                     jobs_list = node.gather_jobs_on_node()
-                if jobs_list is not None:
-                    # Don't clean up the cgroup we just made for the
-                    # job just yet
-                    if jobid not in jobs_list:
-                        jobs_list.append(jobid)
+                except:
+                    pbs.logmsg(pbs.EVENT_DEBUG2,
+                               "Failed to resyncing local job data")
+                # Job list should contain the new jobid. Do not attempt to
+                # cleanup orphaned cgroups with an incomplete job list.
+                # There could be other active jobs missing from the list.
+                if jobs_list and jobid not in jobs_list:
+                    pbs.logmsg(pbs.EVENT_DEBUG2, "Job not found: %s" % jobid)
+                else:
                     self.cleanup_orphans(jobs_list)
+                # Pause after the first attempt
+                if attempt < 1:
+                    sleep(0.5)
         if not assigned:
-            # Cleanup cgroups for jobs not present on this node
+            # Log a message and rerun the job
             pbs.logmsg(pbs.EVENT_DEBUG2,
-                       "Assign resources failed. Attempting to cleanup all " +
-                       "leftover cgroups (including event job %s)" % jobid)
-            if not jobs_list:
-                pbs.logmsg(pbs.EVENT_DEBUG2, "Local job list is empty!")
-            else:
-                # Remove the current job from the jobs list so it will be
-                # cleaned up as well.
-                try:
-                    jobs_list.remove(jobid)
-                    self.cleanup_orphans(jobs_list)
-                except:
-                    pbs.logmsg(pbs.EVENT_DEBUG2,
-                               "Failed to remove job %s" % (jobid))
-                    pass
-            # Rerun the job and log the message
-            line = 'Unable to assign resources to job. '
-            line += 'Will requeue the job and try again. '
-            line += 'job run_count: %d' % pbs.event().job.run_count
+                       'Requeuing job %s' % jobid)
+            pbs.logmsg(pbs.EVENT_DEBUG2,
+                       'Run count for job %s: %d' %
+                       (jobid, pbs.event().job.run_count))
             pbs.event().job.rerun()
-            pbs.event().reject(line)
+            pbs.event().reject('Failed to assign resources')
         # Print out the assigned resources
         pbs.logmsg(pbs.EVENT_DEBUG2,
                    "Assigned resources: %s" % (assigned))
         self.assigned_resources = assigned
         if cpuset_enabled:
-            hostresc.pop('ncpus', None)
+            # Remove the ncpus key if it exists. Ignore any KeyError.
+            try:
+                hostresc.pop('ncpus')
+            except:
+                pass
```

Comment:
This would be shorter if you said if 'ncpus' in hostresc: hostresc.pop('ncpus').  2 lines instead of 4.  Plus if you just want to remove 'ncpus' from hostresc, I think using del hostresc['ncpus'] would be more explicit.  pop returns the value which you are ignoring.